src/ui/common/src/components/pages/workflows/components/ResourceItem.tsx

```diff
@@ -0,0 +1,96 @@
+import { Box, Tooltip, Typography } from '@mui/material';
+import React from 'react';
+
+import { theme } from '../../../../styles/theme/theme';
+import { EngineType, EngineTypeToService } from '../../../../utils/engine';
+import { ServiceLogos } from '../../../../utils/integrations';
+
+export interface ResourceItemProps {
+  resource: string;
+  resourceCustomName?: string; // If included, this will override the default name of the engine that we use.
+  defaultBackgroundColor?: string; // If included, this will override the default background color if none is specified for the particular engine.
+  size?: string; // The font and icon size in pixels.
+  collapseName?: boolean; // If true, the name associated with this engine will be in a tooltip. False by default.
+}
+
+// These two objects are used when we want to show custom definitions for the ResourceItem. Otherwise,
+// we default back to the ServicesLogos imported above.
+const ResourceItemBackgrounds = {
+  [EngineType.Databricks]: '#FF3B29', // The Databricks logo color.
+  [EngineType.K8s]: '#2E6CE6', // The Kubernetes logo color.
+  [EngineType.Lambda]: '#FE7E00', // The Lambda logo color.
+};
+
+const ResourceItemIcons = {
```

Comment:
Something we should probably do here is upload these logos to our aqueduct-public-assets-bucket. You never know when these URLs might change out from under us.